lib/uklibid/libname.c

```diff
@@ -16,3 +19,21 @@ const char *uk_libname(unsigned int libid)
 
 	return namemap[libid];
 }
+
+#if CONFIG_HAVE_LIBC || CONFIG_LIBNOLIBC
+unsigned int uk_libid(const char *libname)
+{
+	unsigned int libid;
+
+	if (unlikely(!libname))
```

Comment:
An assertion here would avoid an unnecessary conditional, even if it's an optimized one

Reply:
actually, right :+1:. There is no good reason to call: `uk_libid(NULL)`